src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AdvSimdHelper.cs

```diff
@@ -0,0 +1,135 @@
+        public static Vector128<sbyte> CreateEscapingMask_UnsafeRelaxedJavaScriptEncoder(Vector128<sbyte> sourceValue)
+        {
+            Debug.Assert(AdvSimd.IsSupported);
+
+            // Anything in the control characters range (except 0x7F), and anything above sbyte.MaxValue but less than or equal byte.MaxValue
+            // That's because anything between 128 and 255 (inclusive) will overflow and become negative.
+            Vector128<sbyte> mask = AdvSimd.CompareLessThan(sourceValue, s_spaceMaskSByte);
+
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_quotationMarkMaskSByte));
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_reverseSolidusMaskSByte));
+
+            // Leftover control character in the ASCII range - 0x7F
+            // Since we are dealing with sbytes, 0x7F is the only value that would meet this comparison.
+            mask = AdvSimd.Or(mask, AdvSimd.CompareGreaterThan(sourceValue, s_tildeMaskSByte));
+
+            return mask;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<sbyte> CreateEscapingMask_DefaultJavaScriptEncoderBasicLatin(Vector128<sbyte> sourceValue)
+        {
+            Debug.Assert(AdvSimd.IsSupported);
+
+            Vector128<sbyte> mask = CreateEscapingMask_UnsafeRelaxedJavaScriptEncoder(sourceValue);
+
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_ampersandMaskSByte));
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_apostropheMaskSByte));
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_plusSignMaskSByte));
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_lessThanSignMaskSByte));
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_greaterThanSignMaskSByte));
+            mask = AdvSimd.Or(mask, AdvSimd.CompareEqual(sourceValue, s_graveAccentMaskSByte));
+
+            return mask;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<short> CreateAsciiMask(Vector128<short> sourceValue)
+        {
+            Debug.Assert(AdvSimd.IsSupported);
+
+            // Anything above short.MaxValue but less than or equal char.MaxValue
+            // That's because anything between 32768 and 65535 (inclusive) will overflow and become negative.
+            Vector128<short> mask = AdvSimd.CompareLessThan(sourceValue, s_nullMaskInt16);
+
+            // Anything above the ASCII range
+            mask = AdvSimd.Or(mask, AdvSimd.CompareGreaterThan(sourceValue, s_maxAsciiCharacterMaskInt16));
+
+            return mask;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ContainsNonAsciiByte(Vector128<sbyte> value)
+        {
+            Debug.Assert(AdvSimd.Arm64.IsSupported);
+            return AdvSimd.Arm64.MinAcross(value).ToScalar() < 0;
```

Comment:
Tagging @TamarChristinaArm to confirm if this will be faster than:

```csharp
// Vector128<byte> mostSignficantBitMask = Vector128.Create((byte)0x80);
// Vector128<byte> mask = Vector128.Create((ushort)0x1001).AsByte();

Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareEqual(AdvSimd.And(value, s_mostSignficantBitMask), s_mostSignficantBitMask);
Vector128<byte> extractedBits = AdvSimd.And(mostSignificantBitIsSet, s_bitMask128);
extractedBits = AdvSimd.Arm64.AddPairwise(extractedBits, extractedBits);
return extractedBits.AsUInt64().ToScalar();
```

Reply:
`SMINV` is to generally be avoided unless no other choice, in this case I believe since you only case about `zero/non-zero`.
So in assembly (I don't remember the C# intrinsics name here) this would be
```
umaxp v1.16b, v0.16b, v0.16b
fmov x0, d1
ands x0, x0, 0x8080808080808080
```
which will be far faster than the `sminv`

Reply:
@TamarChristinaArm, so this would be:
```csharp
var tmp = MaxPairwise(value, value).AsUInt64().ToScalar();
return tmp & 0x8080808080808080;
```

Is that correct?

Reply:
Yup! that's correct @tannergooding

Reply:
Also if you really need the result to be `0` or `1` then the generated sequence should look something like
```
umaxp v1.16b, v0.16b, v0.16b
fmov x0, d1
tst x0, 0x8080808080808080
cset x0, ne
```

which avoids the branch at the end.

Reply:
For signed byte vectors this should probably be:
```csharp
value = AdvSimd.Arm64.MinPairwise(value, value);
return (value.AsUInt64().ToScalar() & 0x8080808080808080) != 0;
```

Reply:
I believe you want to convert `value` to `UInt64` before doing the op, as the point is to compare/add/etc the two 64-bit halves in a single pairwise op.

Reply:
There are no long overloads for Min/MaxPairwise. But I suppose your comment implies that the fewer the lanes of a vector the faster the operation? If that's the case I could perhaps just cast the vector to int32 which does have an overload.

Reply:
I don't think you need to convert `value` to `UInt64`. We want to perform operations on `byte` because that is the representation of original value. However, when we actually check, we just reinterpret them to `ulong`. @eiriktsarpalis  - you should double check that this implementation is functionally correct for boundary cases.